scripts/prepare_sector_network.py

```diff
@@ -946,6 +951,87 @@ def add_storage(n, costs):
         lifetime=costs.at['H2 pipeline', 'lifetime']
     )
 
+    if options["gas_network"]:
+
+        logger.info("Add gas network")
+
+        cols = [
+            "bus0",
+            "bus1",
+            "is_bothDirection",
+            "pipe_capacity_MW",
+            "id",
+            "length_km"
+        ]
+        gas_pipes = pd.read_csv(snakemake.input.clustered_gas_network)[cols]
+        index = "Gas pipeline " + gas_pipes["bus0"] + " -> " + gas_pipes["bus1"]
+        index[gas_pipes.is_bothDirection==1] = index[gas_pipes.is_bothDirection==1].str.replace("->", "<->")
+        gas_pipes.index = index
+
+        # group parallel pipes together
+        strategies = {
+            'bus0': 'first',
+            'bus1': 'first',
+            'is_bothDirection': 'first',
+            "pipe_capacity_MW": 'sum',
+            "length_km": 'sum',
+            'id': ' '.join,
+        }
+        num_parallel = gas_pipes.groupby(gas_pipes.index).count().iloc[:,0]
+        gas_pipes = gas_pipes.groupby(gas_pipes.index).agg(strategies)
+
+        both_direction_i = gas_pipes[gas_pipes.is_bothDirection==1].index
+        # TODO capital cost from DEA energy transport data sheet 102 6
+        # lifetime 50 a, investment ~1 EUR/MW/km, FOM 0.12 EUR/MW/km/year, discountrate 7%
+        pipe_fixed_cost = 0.19   # EUR/MW/km/a (annualised investment costs)
+        gas_pipes["p_min_pu"] = 0.
+        gas_pipes.loc[both_direction_i, "p_min_pu"] = -1
+
+        n.madd("Link",
+            gas_pipes.index,
+            bus0=gas_pipes.bus0 + " gas",
+            bus1=gas_pipes.bus1 + " gas",
+            p_min_pu=gas_pipes.p_min_pu,
+            p_nom=gas_pipes.pipe_capacity_MW,
+            p_nom_extendable=False,
+            length=gas_pipes.length_km,
+            capital_cost=gas_pipes.length_km * pipe_fixed_cost,
+            type=num_parallel,
+            tags=gas_pipes.id,
+            carrier="Gas pipeline",
+            lifetime=50
+        )
+        
+        # remove fossil generators at all connected nodes
+        # TODO what should be assumed here? rather located at LNG terminals?
+        missing = nodes.difference(pd.concat([gas_pipes.bus0, gas_pipes.bus1]).unique())
+        remove_i = n.generators[(n.generators.carrier=="gas")
+                              & (~n.generators.bus.str.replace(" gas","").isin(missing))].index
+        n.generators.drop(remove_i, inplace=True)
```

Comment:
- [ ] Would be good to have the locations of LNG terminals as entry points for fossil gas even for connected sections.

- [ ] Furthermore, one could allow building new gas pipelines to connect isolated notes.

Reply:
@lisazeyen is there processed data for the locations of LNG terminals already or would you just take it from the ENTSO-G capacity map?

Reply:
- LNGs and entries: https://www.gas.scigrid.de/posts/2020-Sep-02_inet_filled.html
- entries: https://www.gas.scigrid.de/posts/2010-Okt-03_inet_raw.html
- LNG terminals: https://www.gas.scigrid.de/posts/2010-Okt-03_gie_raw.html

possibly also in the fully merged dataset: https://www.gas.scigrid.de/posts/2020-Sep-02_iggielgn.html